src/OVAL/oval_variable.c

```diff
@@ -419,6 +523,39 @@ struct oval_variable *oval_variable_clone(struct oval_definition_model *new_mode
 
 			evar = (oval_variable_EXTERNAL_t *) new_variable;
 			old_evar = (oval_variable_EXTERNAL_t *) old_variable;
+
+			struct oval_iterator *old_pv_itr;
+			old_pv_itr = oval_variable_get_possible_values(old_variable);
+			while (oval_collection_iterator_has_more(old_pv_itr)) {
+				struct oval_variable_possible_value *old_pv, *new_pv;
+				old_pv = oval_collection_iterator_next(old_pv_itr);
+				new_pv = oval_variable_possible_value_new(old_pv->hint, old_pv->value);
+				oval_collection_add(evar->possible_values, new_pv);
+			}
+			oval_collection_iterator_free(old_pv_itr);
+
+			struct oval_iterator *old_pr_itr;
+			old_pr_itr = oval_variable_get_possible_restrictions(old_variable);
+			while (oval_collection_iterator_has_more(old_pr_itr)) {
+				struct oval_variable_possible_restriction *old_pr, *new_pr;
+				old_pr = oval_collection_iterator_next(old_pr_itr);
+				new_pr = oval_variable_possible_restriction_new(old_pr->operator, old_pr->hint);
```

Comment:
This allocates *new_pr->restrictions* and then line 546 possibly allocates it again, leaking memory.